xbmc/utils/SystemInfo.cpp

```diff
@@ -134,7 +134,12 @@ static bool appendWindows10NameVersion(std::string& osNameVer)
 #if defined(TARGET_LINUX) && !defined(TARGET_ANDROID)
 static std::string getValueFromOs_release(std::string key)
 {
-  FILE* os_rel = fopen("/etc/os-release", "r");
+#if defined(TARGET_WEBOS)
+  static constexpr const char* releaseFile = "/usr/lib/os-release";
```

Comment:
Might be better to put in an anonymous namespace and drop the static.

This file is also badly in need of splitting into a platform interface. I can look into that at some point.

Reply:
updated